crates/nostr-sdk/src/client/mod.rs

```diff
@@ -206,6 +206,25 @@ impl Client {
         self.keys.clone()
     }
 
+    /// Replace current [`Keys`]
+    ///
+    /// # Example
+    /// ```rust,no_run
+    /// use nostr_sdk::prelude::*;
+    ///
+    /// # #[tokio::main]
+    /// # async fn main() {
+    /// #   let keys1 = Keys::generate();
+    /// #   let keys2 = Keys::generate();
+    /// #   let mut client = Client::new(&keys1);
+    /// #   client.set_keys(&keys2);
+    /// #   assert_eq!(keys2, client.keys());
+    /// # }
+    /// ```
+    pub fn set_keys(&mut self, keys: &Keys) {
+        self.keys = keys.clone()
+    }
+
```

Comment:
Thanks again!
Since the nostr `Client` can be shared across multiple threads, I think this method should not be a mutable.
A solution it's to wrap the `Keys` struct inside a `Arc<RwLock<Keys>>` (where the `RwLock` it's `tokio::sync::RwLock`).
So should be something like:

```rust
    pub async fn set_keys(&self, keys: &Keys) {
        let mut current_keys = self.keys.write().await;
        *current_keys = keys.clone();
    }
```
What do you think?
I expect that are many changes to do to (also in `bindings` crates). Let me know if you prefer that I'll do these changes.

Reply:
That's a good point. I can't think of a better solution than that which you proposed. I wasn't anticipating that this feature would require updates across the library and bindings.

Alternatively I could create a new Client when needed with the new Keys and a clone of the RelayPool. There might be some gotchas related to using the cloned relay pool. What do you think?

Reply:
Yeah, the last proposal could be a solution for some scenarios.

But anyway, I think have a method to allow to change the `Keys` without change client would be useful. 
As I said above, there are no problem if you prefer that I'll work on this change (wrap the keys in `Arc<RwLock<Keys>>`). Just let me know so I'll avoid to work on this if you already working on it.

Reply:
The client is actually really useful so thinking about it a bit more the last solution is not ideal.

I initially thought that your proposed solution would be adding additional complexity to your library but I can see that you have used a similar approach quite a lot. 

> if you prefer that I'll work on this change

That would be great thanks :)

> Just let me know so I'll avoid to work on this if you already working on it.

You said that there is an impact on the bindings. I've not really looked at them so I think you're much better placed to make the change than me.

Once again, thanks very much.

Reply:
Take a look to PR #170 

> You said that there is an impact on the bindings

My bad... yes, but less then expected (just a very little change on bindings).

Reply:
I've just reviewed #170